openshift/e2e-common.sh

```diff
@@ -96,22 +100,162 @@ function install_strimzi(){
   oc apply -f "${KAFKA_USERS_CONFIG}" -n kafka || return 1
 }
 
-function install_serverless(){
-  header "Installing Serverless Operator"
+################################################################################
+# Simple function to do arithmetic calculation using AWK
+#
+# Globals:
+#   None
+# Arguments:
+#   *: Arithmetic expression to calculate (e.g. 3 + 5 , 8/2)
+# Output:
+#   Writes the calculated value to stdout
+################################################################################
+function calc() {
+    awk "BEGIN { print $*; }"
+}
+
+################################################################################
+# Install eventing from midstream repo the matches the current branch
+#
+# The current branch name should match the midstream eventing branch used.
+# Globals:
+#   EVENTING_REPO
+#   EVENTING_NAMESPACE
+# Arguments:
+#   None
+################################################################################
+function install_midstream_eventing() {
+  local branch=$(git rev-parse --abbrev-ref HEAD)
+  header "Installing Knative Eventing from ${branch}"
+  local eventing_dir=/tmp/eventing-operator
+  local failed=0
+  git clone --branch $branch $EVENTING_REPO $eventing_dir || return 1
+  pushd $eventing_dir || return 1
+
+  cat openshift/release/knative-eventing-ci.yaml > ci
+  cat openshift/release/knative-eventing-mtbroker-ci.yaml >> ci
+
+  oc apply -f ci || return 1
+  rm ci
+
+  # Wait for 5 pods to appear first
+  timeout 900 '[[ $(oc get pods -n $EVENTING_NAMESPACE --no-headers | wc -l) -lt 5 ]]' || return 1
+  wait_until_pods_running $EVENTING_NAMESPACE || return 1
+
+  popd || return 1
+  return $failed
+}
+
+################################################################################
+# Calculate the serverless operator release which corresponds to the given
+# eventing version regardless of whether that release exists or not.
+#
+# e.g eventing 0.25 -> 1.19, eventing 1.0 -> 1.21, eventing 1.2 -> 1.23.
+# Globals:
+#   None
+# Arguments:
+#   1: Eventing version number. E.g. 0.26 or 1.2
+# Outputs:
+#   Writes the calculated release version to stdout
+################################################################################
+function calculate_so_release() {
```

Comment:
Noticed this is different than in #524  (for release-next)

shoudl we name it here `calculate_so_release_branch` ?

Reply:
this one is the more accurate one, cause it just calculate the release number "1.21" etc regardless of it being a branch or not.